src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1537,13 +1537,16 @@ private void DoMiscValidation()
             {
                 var diagnostics = BindingDiagnosticBag.GetInstance();
 
-                if (IsParams && ParameterSyntax?.Modifiers.Any(SyntaxKind.ParamsKeyword) == true)
+                bool inExtension = this.ContainingSymbol is NamedTypeSymbol { IsExtension: true };
+                // An error is already reported for `params` in extensions
+                if (IsParams && ParameterSyntax?.Modifiers.Any(SyntaxKind.ParamsKeyword) == true && !inExtension)
                 {
                     validateParamsType(diagnostics);
                 }
 
-                if (DeclaredScope == ScopedKind.ScopedValue && !Type.IsErrorOrRefLikeOrAllowsRefLikeType())
+                if (DeclaredScope == ScopedKind.ScopedValue && !Type.IsErrorOrRefLikeOrAllowsRefLikeType() && (!inExtension || this.Ordinal == 0))
```

Comment:
>this.Ordinal == 0 [](http://example.com/codeflow?start=127&length=17)

I think we shouldn't be creating extension parameters that fail the ordinal check.